week-1-ts-ecosystem/w01e04-light-me-up/src/LightBulb.ts

```diff
@@ -0,0 +1,23 @@
+import { PowerSource } from "./PowerSource";
+
+export class LightBulb {
+  protected readonly powerConsumption = 20;
+  public powerSource: PowerSource;
+
+  constructor(powerSource: PowerSource) {
+    this.powerSource = powerSource;
+  }
+
+  lightUp() {
+    this.powerSource.consume(this.powerConsumption);
+    console.log("Lighting...");
+  }
+
+  lightUpFor(seconds) {
```

Comment:
Tutaj brakuje fizycznego upływu czasu :) co sekunda - to ma faktycznie tak swiecić.

Dodatkowo seconds nie ma typu (implicit `any`).

Ale nie musi tego w ogóle być.
ZObacz - wkręcasz żarówkę - i ona ma świecić. Ona będzie świecić tylko `5` sekund bo tylko tyle jest mocy w `powerSource`
Wiec nie powinieneś ustalać ile ma to świecić tylko obsłużyć błąd jak świecić przestanie.

Reply:
Tutaj mój błąd - to że ma to świeci `n` sekund - wynikało z treści zadania - więc może tak zostać :)